src/parser/postgresparser.cpp

```diff
@@ -556,42 +555,42 @@ std::unique_ptr<AbstractExpression> PostgresParser::ColumnRefTransform(ColumnRef
 }
 
 // Postgres.A_Const -> terrier.ConstantValueExpression
-std::unique_ptr<AbstractExpression> PostgresParser::ConstTransform(A_Const *root) {
+std::unique_ptr<AbstractExpression> PostgresParser::ConstTransform(ParseResult *parse_result, A_Const *root) {
   if (root == nullptr) {
     return nullptr;
   }
-  return ValueTransform(root->val_);
+  return ValueTransform(parse_result, root->val_);
 }
 
 // Postgres.FuncCall -> terrier.AbstractExpression
-std::unique_ptr<AbstractExpression> PostgresParser::FuncCallTransform(FuncCall *root) {
-  // TODO(WAN): change case?
+std::unique_ptr<AbstractExpression> PostgresParser::FuncCallTransform(ParseResult *parse_result, FuncCall *root) {
+  // TODO(WAN): Check if we need to change the case of this.
   std::string func_name = reinterpret_cast<value *>(root->funcname_->head->data.ptr_value)->val_.str_;
 
   std::unique_ptr<AbstractExpression> result;
   if (!IsAggregateFunction(func_name)) {
     // normal functions (built-in functions or UDFs)
     func_name = (reinterpret_cast<value *>(root->funcname_->tail->data.ptr_value))->val_.str_;
-    std::vector<std::shared_ptr<AbstractExpression>> children;
+    std::vector<std::unique_ptr<AbstractExpression>> children;
 
     if (root->args_ != nullptr) {
       for (auto cell = root->args_->head; cell != nullptr; cell = cell->next) {
         auto expr_node = reinterpret_cast<Node *>(cell->data.ptr_value);
-        children.emplace_back(ExprTransform(expr_node));
+        children.emplace_back(ExprTransform(parse_result, expr_node, nullptr));
       }
     }
     result = std::make_unique<FunctionExpression>(func_name.c_str(), type::TypeId::INVALID, std::move(children));
   } else {
     // aggregate function
     auto agg_fun_type = StringToExpressionType("AGGREGATE_" + func_name);
-    std::vector<std::shared_ptr<AbstractExpression>> children;
+    std::vector<std::unique_ptr<AbstractExpression>> children;
     if (root->agg_star_) {
       auto child = std::make_unique<StarExpression>();
       children.emplace_back(std::move(child));
       result = std::make_unique<AggregateExpression>(agg_fun_type, std::move(children), root->agg_distinct_);
     } else if (root->args_->length < 2) {
       auto expr_node = reinterpret_cast<Node *>(root->args_->head->data.ptr_value);
-      auto child = ExprTransform(expr_node);
+      auto child = ExprTransform(parse_result, expr_node, nullptr);
```

Comment:
this isn't being explicitly being casted so try to avoid using auto so the type is more evident

Reply:
I'm not particularly convinced, given that everyone is developing in CLion and the type is a control-mouseover away.. the code is messy enough already. There's also a lot of places that this comment could apply, and I'd rather keep it consistently terse (a reader is expected to remember what ExprTransform returns) than to have it in some places / all places.

Though I'm not hugely opposed to it either.